plugins/modules/proxmox_cluster.py

```diff
@@ -0,0 +1,302 @@
+  returned: when action == 'get_cluster_info'
+  type: dict
+  contains:
+    config_digest:
+      description: Digest of the cluster configuration.
+      type: str
+      sample: "aef68412f7976505ed083e6173b96274a281da25"
+    nodelist:
+      description: List of nodes in the cluster.
+      type: list
+      elements: dict
+      contains:
+        name:
+          description: Node name.
+          type: str
+          sample: "pve2"
+        nodeid:
+          description: Node ID.
+          type: str
+          sample: "1"
+        pve_addr:
+          description: Proxmox VE address.
+          type: str
+          sample: "10.10.10.159"
+        pve_fp:
+          description: Proxmox VE fingerprint.
+          type: str
+          sample: "08:B5:B2:F9:EC:01:0B:D0:..."
+        quorum_votes:
+          description: Quorum votes assigned to the node.
+          type: str
+          sample: "1"
+        ring0_addr:
+          description: Address for ring0.
+          type: str
+          sample: "vmbr0"
+    preferred_node:
+      description: The preferred cluster node.
+      type: str
+      sample: "pve2"
+    totem:
+      description: Totem protocol configuration.
+      type: dict
+      contains:
+        cluster_name:
+          description: Cluster name from totem.
+          type: str
+          sample: "devcluster"
+        config_version:
+          description: Config version.
+          type: str
+          sample: "1"
+        interface:
+          description: Interface configuration.
+          type: dict
+        ip_version:
+          description: IP version.
+          type: str
+          sample: "ipv4-6"
+        link_mode:
+          description: Link mode.
+          type: str
+          sample: "passive"
+        secauth:
+          description: Whether secure authentication is on.
+          type: str
+          sample: "on"
+        version:
+          description: Totem protocol version.
+          type: str
+          sample: "2"
+"""
+
+import traceback
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import missing_required_lib
+
+try:
+    from proxmoxer import ProxmoxAPI
+except ImportError:
+    PROXMOXER_LIBRARY = False
+    PROXMOXER_LIBRARY_IMPORT_ERROR = traceback.format_exc()
+else:
+    PROXMOXER_LIBRARY = True
+    PROXMOXER_LIBRARY_IMPORT_ERROR = None
+
+
+def create_cluster(proxmox, module):
+    cluster_name = module.params.get("cluster_name") or module.params.get("api_host")
+    payload = {"clustername": cluster_name}
+
+    if module.params.get("link0") is not None:
+        payload["link0"] = module.params.get("link0")
+    if module.params.get("link1") is not None:
+        payload["link1"] = module.params.get("link1")
+
+    if module.check_mode:
+        module.exit_json(changed=True, msg="Cluster '{}' would be created (check mode).".format(cluster_name), cluster=cluster_name)
+
+    try:
```

Comment:
How does this handle when creating a cluster that already exists? Will it error out, or just tell the user it's all fine and return 'ok'?